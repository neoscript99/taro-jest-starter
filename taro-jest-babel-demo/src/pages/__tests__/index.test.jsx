import Nerv from 'nervjs'
import { renderToString } from 'nerv-server'
import { Provider } from '@tarojs/mobx'
import Index from '../../../.temp/pages/index'
import IndexPure from '../../../.temp/pages/index/pure'
import counterStore from '../../store/counter'

const store = {
  counterStore
}

describe('ui test', () => {
  it('no mobx', () => {
    const component = renderToString(
      <IndexPure counterStore={counterStore} />
    )
    expect(component)
      .toMatchSnapshot()
  })

  it('with mobx', () => {
    const component = renderToString(
      <Provider store={store}>
        <Index />
      </Provider>
    )
    expect(component)
      .toMatchSnapshot()
  })
})
