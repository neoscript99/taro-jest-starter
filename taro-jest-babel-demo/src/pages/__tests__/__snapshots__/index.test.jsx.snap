// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui test no mobx 1`] = `"<div class=\\"index\\"><button class=\\"weui-btn weui-btn_default\\">undefined+</button><button class=\\"weui-btn weui-btn_default\\">undefined-</button><button class=\\"weui-btn weui-btn_default\\">undefinedAdd Async</button><span class=\\"taro-text\\">0</span></div>"`;
